### DISCUSSION TITLE
Updating materialized views when data goes away

### DISCUSSION THREAD

**Discussion by wdiechmann** [scenic-views/scenic#408] [Q&A] [answered]
This question may be a total 'n00b' one - but here goes anyway:

I'm using materialized views with Scenic and being quite happy with it too!

My app is multi tenant-ish and I'm allowing users a free trial. Some may leave, and then I have to 'clean' up after - deleting records right and left. I've got a background job running to refresh my materialized views 

```

  def self.refresh
    Scenic.database.refresh_materialized_view(table_name, concurrently: false, cascade: false)
  end

```

and all is good and well.

_But I'm not sure if the cached tables are being emptied of data belonging to accounts deleted?_

(If this all makes no sense what so ever please accept my apologies)

**Comment by calebhearth** [comment, marked as answer]
Good morning, Walther. Yes the mat view should not include deleted rows from tables (or views) that no longer contain the row. Only one the refresh has happened or course.

**Comment by wdiechmann** [comment]
thank you so much @calebhearth - now I just need to find the culprit making this _not_ happening (but that's a lot easier once you know what to expect)

 😉 

```
test "the table should be empty" do 
  AccountDeleteJob.new.perform accounts(:one)
  EuConformance.refresh
  assert_equal 0, EuConformance.all.count
end
```
